### ISSUE TITLE
DynamicForm inside element with *ngIf throws synchronization error

### ISSUE THREAD

**Issue by minuz** [udos86/ng2-dynamic-forms#266]
Hi @udos86,

Firstly, great job with this component! It saved me days of work :D 
Secondly, let me describe the issue.

Initially I had this form within a modal that gets triggered by a click that switch the ngIf
When the ngIf=true and the inner content of the component starts to load, the dynamicForm throws the following: 

![image](https://cloud.githubusercontent.com/assets/818024/22767817/fdfe08ca-eed1-11e6-9565-0123d03b7786.png)

So after some debugging, I finally narrowed down to the ngIf, which in its simplest form, the markup would be like the following:

``` HTML
<div class="col-md-12">
    <fieldset>
        <legend class="text-semibold">
            <i class="icon-list-unordered position-left"></i>Table Columns
            <a (click)="toggleView()" class="btn btn-default text-capitalize pull-right mt-10"><i class="icon-add-to-list"></i></a>
        </legend>
        <field-list *ngIf="display" [fields]="fields"></field-list>
        <table-columns-form *ngIf="!display"></table-columns-form>
    </fieldset>
</div>
```

the Component: 

``` JavaScript
export class CustomTypesFieldsFormComponent implements OnInit {

    display: boolean;
    fields: any[];

    constructor(private customTypeService: CustomTypeService) { }

    ngOnInit() {
        this.fields = [];
    }

    toggleView() {
        this.display = !this.display;
    }

    onModalCancel(e: any) {
        this.toggleView();
    }

    onModalSubmit(e: any) {
        this.fields.push(e.value);
        this.fieldFormSubmit.emit(e);
        this.toggleView();
    }
}

```

the DynamicFormComponent:

``` JavaScript
@Component({
    moduleId: module.id,
    selector: 'table-columns-form',
    template: `
    <form class="form-horizontal" [formGroup]="form">
        <div class="panel panel-flat no-border no-margin no-shadow">
            <div class="panel-body no-padding">
                <div class="row">
                    <dynamic-form-bootstrap-control *ngFor="let controlModel of formModel" [controlGroup]="form" [model]="controlModel">
                    </dynamic-form-bootstrap-control>
                </div>
            </div>
        </div>
    </form>
    `
})
export class CustomTypesColumnsFormComponent implements OnInit {

    formModel: Array<DynamicFormControlModel> = COLUMN_DEFINITION_FORM_MODEL;
    form: FormGroup;

    constructor(private svc: DynamicFormService) { }

    ngOnInit() {
        this.form = this.svc.createFormGroup(this.formModel);
    }
}
```

I hope I'm missing something stupid...
but at the moment, if I remove the ngIf, it loads the DynamicForm correctly, with the ngIf it throws that error.

Can you have a look on this please? I've been trying to build the example, but It has some issues with the optional dependency on KendoUI.

Really appreciate any help.
Cheers, 
Fernando

**Comment by minuz** [comment]
I made a fork of your project and added a demonstration of the bug.
https://github.com/LucsanCapital/ng2-dynamic-forms/commit/9eac83e546ea6afb321f3254b6d8d3f833f9a5e7

**Comment by udos86** [comment]
@minuz Thanks for the kind words! This is a [very well known](https://github.com/angular/angular/issues/6005) exception in Angular 2. It is thrown only in DevMode when a binding has changed after initialization while Angular's change detection has not been triggered anew.

I am not sure why it occurs in your sample code. You could ultimately avoid by activating ProdMode. However I'll investigate your bug and will let you know as soon as possible.

**Comment by udos86** [comment]
@minuz After some further investigation I can now tell you that the error is caused by `DynamicFormArrayModel`. I'll try to deliver a fix!

**Comment by udos86** [comment]
@minuz Please check out latest patch version `1.3.12`. Hopefully this should fix your problem.

**Comment by minuz** [comment]
Just for future reference, there's another way to fix this problem.
On my component, I modified the changeDetection to ChangeDetectionStrategy.OnPush 
This basically fixed the issue without having to touch your code.

I saw the change you did for this. I'll check it out if works as soon as I can.

**Comment by minuz** [comment]
FYI, this change did not fix the issue for me. On my component I still have to change the `changeDetection` strategy for the component to `ChangeDetectionStrategy.OnPush`

**Comment by udos86** [comment]
@minuz I can assure you that I made the fix based on your commit LucsanCapital@9eac83e. I'll have another look.

**Comment by udos86** [comment]
@minuz Could you please switch to `development` branch in your fork and pull the latest commit f588aab? I've excluded Kendo from the example so that you can easily try out your bug demo. There should be no exeception thrown with `1.3.12`.
